### PR TITLE
expand compressed Application global data into A5 World

### PR DESCRIPTION
**Pull request by gbody**

### CODE REVIEW

**Comment by uxmal** [comment]
Thanks for the contribution, @gbody. I will review and comment the PR now.

File: src/Environments/MacOS/ResourceFork.cs [hunk 1]
```diff
@@ -342,7 +343,115 @@ public void Dump()
...
+                            }
+                        }
+                    }
+                }
+                    
+            }
+        }
+
+        private int GetRL(BeImageReader a5dr, ref int a5rpt)
+        {
```
Comment (uxmal) [inline src/Environments/MacOS/ResourceFork.cs]: We can afford the longer name `GetRunLength` here.

File: src/Environments/MacOS/ResourceFork.cs [hunk 2]
```diff
@@ -342,7 +343,115 @@ public void Dump()
...
+            int RL = a5dr.ReadByte();
```
Comment (uxmal) [inline src/Environments/MacOS/ResourceFork.cs]: Consider adhering to the coding style: local variables are in camel case. We can afford a longer, more self-explaining name: `runLength`.

File: src/Environments/MacOS/ResourceFork.cs [hunk 3]
```diff
@@ -342,7 +343,115 @@ public void Dump()
...
+                                a5dr.Seek(a5dhdroffset - 2);
+                                var a5dbelow = a5dr.ReadBeUInt32();
+                                var a5dbanksize = a5dr.ReadBeUInt32();
+                                var a5doffset = a5dr.ReadBeUInt32();
+                                var a5dreloc = a5dr.ReadBeUInt32();
+                                var a5dhdrend = a5dr.ReadBeUInt32();
+                                if (a5dbanksize == 0x00010000)
+                                {
+                                    var a5worldbaseoffset = a5world.MemoryArea.BaseAddress + jt.BelowA5Size;
+                                    var a5beloww = new BeImageWriter(a5world.MemoryArea, jt.BelowA5Size - a5dbelow);
```
Comment (uxmal) [inline src/Environments/MacOS/ResourceFork.cs]: Suggest calling this variable `a5belowWriter` to make it clearer what it is used for.

File: src/Environments/MacOS/ResourceFork.cs [hunk 4]
```diff
@@ -342,7 +343,115 @@ public void Dump()
                 platform.A5World = a5world;
                 platform.A5Offset = jt.BelowA5Size;
                 segmentMap.AddSegment(a5world);
+
+                // Check for %A5Init CODE Segment and unpack global data to A5World
+                foreach ( var a5dataseg  in segmentMap.Segments.Values)
+                {
+                    if (a5dataseg.Name.Length >= 12 )
```
Comment (uxmal) [inline src/Environments/MacOS/ResourceFork.cs]: These deeply nested `if`s and `do` make it hard to follow the logic. Consider switching the logic around so that if either of the two criteria doesn't fit, short circuit out of the `foreach` with a continue.

File: src/Environments/MacOS/ResourceFork.cs [hunk 5]
```diff
@@ -45,6 +45,7 @@ public class ResourceFork
         uint rsrcMapOff;
         uint dataSize;
         uint mapSize;
+        private object a5dr;
```
Comment (uxmal) [inline src/Environments/MacOS/ResourceFork.cs]: This looks like an unused field?

**Comment by uxmal** [comment]
Another comment @gbody : if you want to discuss this in a more "prompt" medium, try using https://gitter.im/uxmal/reko

**Comment by ptomin** [comment]
@gbody Strange commits. There are two same commits with name "A5World global data" and two unnecessary commits. Could you fix it on your local machine and than push with `force` option?

**Comment by gbody** [comment]
Novice issues and looks like I totally wrecked my local repository. I am just going to close this push and clean things up and start again.

**Comment by uxmal** [comment]
OK. If I can be of any assistance, please don't hesitate to ask. We can use whatever medium works best for you, but I suggest joining the Reko gitter.im forum as a first step. https://gitter.im/uxmal/reko.
